source/Panel.h

```diff
@@ -122,9 +132,20 @@ class Panel {
 		std::function<void()> fun;
 	};
 
+	bool DoKeyDown(SDL_Keycode key, Uint16 mod, const Command &command, bool isNewPress);
```

Comment:
This class now has KeyDown and DoKeyDown functions, but it isn't clear to me from the comments around those functions on why we need both. Can you add some comments to the code on why we have both those variants?

Reply:
The UI class no longer directly calls the panel virtual event methods, as the panels can now have child panels instead. The Do* methods are non-virtual methods used by the UI class to propagate the event to the correct child if needed. 

I've added a comment clarifying this to the code.